src/time.rs

```diff
@@ -273,26 +269,40 @@ impl Time {
     /// assert_eq!(d.to_string(), "01:02:20.000123");
     /// ```
     pub fn from_timestamp_with_config(
-        timestamp_second: u32,
+        timestamp: i64,
         timestamp_microsecond: u32,
         config: &TimeConfig,
     ) -> Result<Self, ParseError> {
-        let mut second = timestamp_second;
-        let mut microsecond = timestamp_microsecond;
-        if microsecond >= 1_000_000 {
-            second = second
-                .checked_add(microsecond / 1_000_000)
-                .ok_or(ParseError::TimeTooLarge)?;
-            microsecond %= 1_000_000;
-        }
-        if second >= 86_400 {
+        let (seconds, additional_microseconds) = match config.timestamp_interpretation {
+            TimestampInterpretation::Auto => {
+                if timestamp.abs() > 20_000_000_000 {
+                    (timestamp / 1000, ((timestamp % 1000) * 1000) as u32)
+                } else {
+                    (timestamp, 0)
+                }
+            }
+            TimestampInterpretation::AlwaysSeconds => (timestamp, 0),
+        };
```

Comment:
I think here we don't need to handle this case, as the limit of valid input is 86,400 seconds, i.e 86,400,000 milliseconds which is less than 20,000,000,000.

So my reading is that valid input here will always be treated as seconds anyway.

(A comment saying why we don't use `timestamp_interpretation` here might be useful.)

Reply:
You are right @davidhewitt, I will fix that